adapters/repos/db/vector/hnsw/deserializer.go

```diff
@@ -200,7 +205,14 @@ func (c *Deserializer) ReadLink(r io.Reader, res *DeserializationResult) error {
 	}
 
 	if res.Nodes[int(source)] == nil {
-		res.Nodes[int(source)] = &vertex{id: source, connections: make(map[int][]uint64)}
+		res.Nodes[int(source)] = &vertex{id: source, connections: make([][]uint64, level+1)}
+	}
+
+	if len(res.Nodes[int(source)].connections) <= int(level) {
```

Comment:
Total nitpick, but you have the comparison exactly the other way around than above `int(level) >= len(res.Nodes[id].connections)` vs `len(res.Nodes[int(source)].connections) <= int(level)`

Reply:
This has become obsolete because of the refactor you suggested in the other comment.